packages/@uppy/core/src/index.js

```diff
@@ -593,47 +593,40 @@ class Uppy {
    *
    * The `files` value is passed in because it may be updated by the caller without updating the store.
    */
-  checkAndCreateFileStateObject (files, f) {
-    const fileType = getFileType(f)
-    let file = f
-    file.type = fileType
-
-    const onBeforeFileAddedResult = this.opts.onBeforeFileAdded(file, files)
-
-    if (onBeforeFileAddedResult === false) {
-      // Don’t show UI info for this error, as it should be done by the developer
-      this.showOrLogErrorAndThrow(new RestrictionError('Cannot add the file because onBeforeFileAdded returned false.'), { showInformer: false, file })
-    }
-
-    if (typeof onBeforeFileAddedResult === 'object' && onBeforeFileAddedResult) {
-      file = onBeforeFileAddedResult
-    }
+  checkAndCreateFileStateObject (files, fileDescriptor) {
+    const fileType = getFileType(fileDescriptor)
 
     let fileName
-    if (file.name) {
-      fileName = file.name
+    if (fileDescriptor.name) {
+      fileName = fileDescriptor.name
     } else if (fileType.split('/')[0] === 'image') {
       fileName = `${fileType.split('/')[0]}.${fileType.split('/')[1]}`
     } else {
       fileName = 'noname'
     }
     const fileExtension = getFileNameAndExtension(fileName).extension
-    const isRemote = file.isRemote || false
-
-    const fileID = generateFileID(file)
+    const isRemote = fileDescriptor.isRemote || false
```

Comment:
I find the following more readable, please disregard if you disagree.
```suggestion
    const isRemote = Boolean(fileDescriptor.isRemote)
```

Reply:
💯